cmds/core/sync/sync_linux.go

```diff
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !plan9
-// +build !plan9
+//go:build linux
```

Comment:
but sync is a universal Unix thing sync about 1971. I don't think this is how we want to do this. an option is to make the syscall on line 56 conditioned on runtime.GOOS
if runtime.GOOS=="linux" {
etc.
}

but this command should be available for all unix-like systems.

Reply:
also, in case it is a concern, Go linker does the same kind of code removal the plan 9 compilers do (since it came from there, even in the rewrite, they maintain that ability)

so given this code:
```
func main() {
	if runtime.GOOS == "linux" {
		panic("linux")
	}

	println("hi")
}
```

objdump on linux shows this:
```
TEXT main.main(SB) /Users/r.minnich/tmp/a.go
  a.go:5                0x67720                 f9400b90                MOVD 16(R28), R16
  a.go:5                0x67724                 eb3063ff                CMP R16, RSP
  a.go:5                0x67728                 54000149                BLS 10(PC)
  a.go:5                0x6772c                 f81e0ffe                MOVD.W R30, -32(RSP)
  a.go:5                0x67730                 f81f83fd                MOVD R29, -8(RSP)
  a.go:5                0x67734                 d10023fd                SUB $8, RSP, R29
  a.go:7                0x67738                 b0000060                ADRP 53248(PC), R0
  a.go:7                0x6773c                 91160000                ADD $1408, R0, R0
  a.go:7                0x67740                 f0000161                ADRP 192512(PC), R1
  a.go:7                0x67744                 91278021                ADD $2528, R1, R1
  a.go:7                0x67748                 97ff519a                CALL runtime.gopanic(SB)
  a.go:7                0x6774c                 d503201f                NOOP
  a.go:5                0x67750                 aa1e03e3                MOVD R30, R3
  a.go:5                0x67754                 97fff64f                CALL runtime.morestack_noctxt.abi0(SB)
  a.go:5                0x67758                 17fffff2                JMP main.main(SB)
  a.go:5                0x6775c                 00000000                ?
```

whereas, on darwin
```
  a.go:5                0x100056840             f9400b90                MOVD 16(R28), R16
  a.go:5                0x100056844             eb3063ff                CMP R16, RSP
  a.go:5                0x100056848             540001a9                BLS 13(PC)
  a.go:5                0x10005684c             f81e0ffe                MOVD.W R30, -32(RSP)
  a.go:5                0x100056850             f81f83fd                MOVD R29, -8(RSP)
  a.go:5                0x100056854             d10023fd                SUB $8, RSP, R29
  a.go:10               0x100056858             97ff599e                CALL runtime.printlock(SB)
  a.go:10               0x10005685c             90000000                ADRP 0(PC), R0
  a.go:10               0x100056860             912be000                ADD $2808, R0, R0
  a.go:10               0x100056864             b24007e1                ORR $3, ZR, R1
  a.go:10               0x100056868             97ff5bc6                CALL runtime.printstring(SB)
  a.go:10               0x10005686c             97ff59b9                CALL runtime.printunlock(SB)
  a.go:11               0x100056870             a97ffbfd                LDP -8(RSP), (R29, R30)
  a.go:11               0x100056874             910083ff                ADD $32, RSP, RSP
  a.go:11               0x100056878             d65f03c0                RET
  a.go:5                0x10005687c             aa1e03e3                MOVD R30, R3
  a.go:5                0x100056880             97fff610                CALL runtime.morestack_noctxt.abi0(SB)
  a.go:5                0x100056884             17ffffef                JMP main.main(SB)
```

note that on linux, there is no call to printstring, since the panic will ensure the printstring is not reached.

on darwin, there is no call to panic, since runtime.GOOS will be linux.

So you're quite safe having a conditional on runtime.GOOS, and "the right thing" will happen. 

Note this is the kind of thing that in Rust/C/C++ requires an entire other mechanism (macro); on Go you get it for free, and it just looks like ordinary go code.

Reply:
Yeah, I agree that sync should not be only for linux 
But if we do runtime.GOOS thing we can't use constant `unix.SYS_SYNCFS` because it will not compile on any system other than linux. Right now this command will compile only on linux. 
We can do something like this:
`
if runtime.GOOS=="linux" {
    	return doSyscall(267, args)
}
`
but probably I would prefer to have to os dependent implementation for doSyscall, one for linux and one for unix.

Reply:
I'm ok with us defining that constant, because linux has its stable ABI guarantee, unlike other kernels. 

So if you want to define 
const SYNCFS=267
and use it that works for me. 

We have to allow sync to exist on BSDs at least, so tagging the command with linux build tag is really not going to work out.